### PR TITLE
ENH: Fix multiple compiler warnings and other small issues

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

**Review by JDuffeyBQ — APPROVED** [review]

**Review by mmarineBlueQuartz — APPROVED** [review]

File: src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp [hunk 1]
```diff
@@ -882,6 +892,38 @@ Result<IDataArray*> readLegacyDataArray(DataStructure& dataStructure, const nx::
     return MakeErrorResult<IDataArray*>(error.code, error.message);
   }
 
+  // Let's do some sanity checking of the tuple dims and the comp dims
+  // This is here because some 3rd party apps are trying to write .dream3d
+  // files but are getting confused about the Tuple Dims and the component
+  // dims. This bit of code will try to fix the specific issue where the
+  // .dream3d file essentially had _all_ the dimensions shoved in the
+  // tuple dimensions.
+  if(!amTDims.empty()) // If the amTDims are empty we are just reading a Data Array without an Attribute Matrix so skip.
```
Comment (JDuffeyBQ) [inline src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp:901]: If we detect this issue, we should emit a warning.

**Review by JDuffeyBQ — CHANGES_REQUESTED** [review]
We should add an issue for adding a test case for incorrect dimensions in a legacy DREAM3D file.

**Comment by imikejackson** [comment]
https://github.com/BlueQuartzSoftware/simplnx/issues/1117

File: src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp (hunk 1 above)
Comment (imikejackson) [reply src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp:901]: Done.

File: src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp [hunk 2]
```diff
@@ -883,6 +893,44 @@ Result<IDataArray*> readLegacyDataArray(DataStructure& dataStructure, const nx::
...
+        if(idx >= amTDims.size())
+        {
+          cDims.push_back(tDims[idx]);
+        }
+        else if(amTDims[idx] != tDims[idx]) // something went wrong here. Each vector should have the same exact values at the start of the vector
+        {
+          // This should not happen at all. If so, bail out now.
+          cDims.clear(); // just put things back the way they were and fail.
+          cDims.push_back(1ULL);
+          daResult.errors().push_back({-9545, fmt::format(".dream3d file Tuple and Component Dimensions are not correct. DataSet {} does not have correct component dimensions of they are missing. An "
```
Comment (JDuffeyBQ) [inline src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp]: Should we just return here?

File: cmake/CopyDataFile.cmake.in [hunk 3]
```diff
@@ -1,7 +1 @@
-# --------------------------------------------------------------------------------------------------
-# message(STATUS "[DATA COPY] Copy data from @ARGS_DREAM3D_DATA_DIR@/TestFiles/@ARCHIVE_BASE_NAME@")
-# --------------------------------------------------------------------------------------------------
-# execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory "@ARGS_DREAM3D_DATA_DIR@/TestFiles/@ARCHIVE_BASE_NAME@" "Data/@ARCHIVE_BASE_NAME@"     
-#                    COMMAND_EXPAND_LISTS
-#                    VERBATIM
-#                )
+
```
Comment (JDuffeyBQ) [inline cmake/CopyDataFile.cmake.in]: Is this file still used?

File: src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp (hunk 2 above)
Comment (imikejackson) [reply src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp]: Code added to return when the error hits.

File: cmake/CopyDataFile.cmake.in (hunk 3 above)
Comment (imikejackson) [reply cmake/CopyDataFile.cmake.in]: File was removed.

**Review by JDuffeyBQ — APPROVED** [review]

**Pull request merged**